### PR TITLE
⏪ Claim reward flow

### LINKED ISSUE #3053: Claim Channel Payment Flow
## Context
Originating Epic

* https://github.com/Joystream/atlas/issues/2401

## Design
Link to Figma > [LINK](https://www.figma.com/file/ZFRXsyGvnBXnkDUUGc5vyR/My-payments?node-id=1547%3A19340)

## Scope
* [ ] Claim CTA
* [ ] Modal to review fees
* [ ] Confirmation modal

## Out of scope
* [x] Insufficient Joy to cover tx fees/ null claim / claim more than balance (these do not apply based on max amount claimable at all times)

https://github.com/Joystream/atlas/issues/3086

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202078489349362/1202701542805374) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by WRadoslaw**
fixes #3053

### CODE REVIEW

File: packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+              {formatNumber(availableAward ?? 0)}
+            </Text>
+          </TextWrapper>
+          <NumberFormat
+            variant="t100"
+            as="p"
+            format="dollar"
+            value={convertHapiToUSD(rawHapiAward ?? new BN(0)) ?? 0}
+            margin={{ top: 1 }}
+          />
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx]: Let's hide information about the price in USD until we support it

File: packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+interface ClaimChannelPaymentsDialogProps {
+  show?: boolean
+  onExit?: () => void
+}
+
+export const ClaimChannelPaymentsDialog = ({ onExit, show }: ClaimChannelPaymentsDialogProps) => {
+  const { channelId } = useUser()
+  const { channel } = useFullChannel(channelId || '')
+  const { availableAward, claimReward, rawHapiAward, isAwardLoading, txParams } = useChannelPayout(onExit)
+  const { fullFee, loading: feeLoading } = useFee('claimRewardTx', txParams)
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx:25]: I'm pretty sure that fee isn't correctly calculated. When I logged txParams I got `undefined`

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          </Text>
+        </TextWrapper>
+      )}
+
+      <BalanceWrapper>
+        <Text variant="t100" as="p" color="colorText" margin={{ top: 1 }}>
+          Membership account balance
+        </Text>
+        <Text variant="t100" as="p" color="colorText" margin={{ top: 1 }}>
+          {formatNumber(channelBalance ? hapiBnToTokenNumber(channelBalance) : 0)} tJOY
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx]: Sorry, I missed one thing. This value should be **the member balance**, not **channel balance**. The purpose of displaying that is to make sure, that the user has money to pay the transaction fee. So, let's use here `totalBalance`, ie. `lockedBalance + accountBalance`. 
Like we have here: https://github.com/Joystream/atlas/pull/3454/files#diff-15493a7d81d14bc68ede4728b4b6f1bb65ce118111dc95f7e8fd1c4bace1ff63 (we don't have this fix on payments branch, just FYI)

File: packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            {formatNumber(availableAward ?? 0)}
+          </Text>
+        </TextWrapper>
+      )}
+
+      <BalanceWrapper>
+        <Text variant="t100" as="p" color="colorText" margin={{ top: 1 }}>
+          Membership account balance
+        </Text>
+        <Text variant="t100" as="p" color="colorText" margin={{ top: 1 }}>
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx]: We should use here NumberFormat component instead

File: packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      primaryButton={{ text: 'Claim', onClick: claimReward }}
+      secondaryButton={{ text: 'Cancel', onClick: onExit }}
+      additionalActionsNode={<Fee amount={fullFee} loading={feeLoading} variant="h200" />}
+    >
+      {isAwardLoading ? (
+        <SkeletonLoader height={mdMatch ? 31 : 24} width={128} bottomSpace={5} />
+      ) : (
+        <TextWrapper>
+          <SvgJoyTokenMonochrome24 />
+          <Text variant={mdMatch ? 'h500' : 'h400'} as="p">
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/ClaimChannelPaymentsDialog/ClaimChannelPaymentsDialog.tsx]: Same as above, I think

**Review by drillprop — CHANGES_REQUESTED** [review]

**Comment by WRadoslaw** [comment]
I noticed that there is an error from PR implementing the`useChannelPayout` hook. In `handleTransaction` `cumulativeRewardClaimed` was passed instead of `cumulativeRewardEarned`.

After the fix, the error occurred in OverviewTab, it looks like we have to validate the amount before displaying Claim button (or disable it with a tooltip informing what's going on).

image

**Review by drillprop — APPROVED** [review]
> I noticed that there is an error from PR implementing the`useChannelPayout` hook. In `handleTransaction` `cumulativeRewardClaimed` was passed instead of `cumulativeRewardEarned`.
> 
> After the fix, the error occurred in OverviewTab, it looks like we have to validate the amount before displaying Claim button (or disable it with a tooltip informing what's going on).
> 
> image

I created #3495 to fix this in the future.

Besides that, looks good to me!

**Pull request merged**